boards/litex_sim/src/main.rs

```diff
@@ -0,0 +1,420 @@
+        litex_vexriscv::timer::LiteXAlarm::new(timer0_uptime, timer0)
+    );
+    timer0.set_timer_client(litex_alarm);
+    litex_alarm.initialize();
+
+    // Create a shared virtualization mux layer on top of a single hardware
+    // alarm.
+    let mux_alarm = static_init!(
+        MuxAlarm<
+            'static,
+            litex_vexriscv::timer::LiteXAlarm<
+                'static,
+                'static,
+                socc::SoCRegisterFmt,
+                socc::ClockFrequency,
+            >,
+        >,
+        MuxAlarm::new(litex_alarm)
+    );
+    litex_alarm.set_alarm_client(mux_alarm);
+
+    // Userspace alarm driver
+    let virtual_alarm_user = static_init!(
+        VirtualMuxAlarm<
+            'static,
+            litex_vexriscv::timer::LiteXAlarm<
+                'static,
+                'static,
+                socc::SoCRegisterFmt,
+                socc::ClockFrequency,
+            >,
+        >,
+        VirtualMuxAlarm::new(mux_alarm)
+    );
+    let alarm = static_init!(
+        capsules::alarm::AlarmDriver<
+            'static,
+            VirtualMuxAlarm<
+                'static,
+                litex_vexriscv::timer::LiteXAlarm<
+                    'static,
+                    'static,
+                    socc::SoCRegisterFmt,
+                    socc::ClockFrequency,
+                >,
+            >,
+        >,
+        capsules::alarm::AlarmDriver::new(
+            virtual_alarm_user,
+            board_kernel.create_grant(&memory_allocation_cap)
+        )
+    );
+    virtual_alarm_user.set_alarm_client(alarm);
+
+    // Systick virtual alarm for scheduling
+    let systick_virtual_alarm = static_init!(
+        VirtualMuxAlarm<
+            'static,
+            litex_vexriscv::timer::LiteXAlarm<
+                'static,
+                'static,
+                socc::SoCRegisterFmt,
+                socc::ClockFrequency,
+            >,
+        >,
+        VirtualMuxAlarm::new(mux_alarm)
+    );
+
+    // ---------- UART ----------
+
+    // Initialize the HW UART
+    let uart0 = static_init!(
+        litex_vexriscv::uart::LiteXUart<socc::SoCRegisterFmt>,
+        litex_vexriscv::uart::LiteXUart::new(
+            StaticRef::new(
+                socc::CSR_UART_BASE
+                    as *const litex_vexriscv::uart::LiteXUartRegisters<socc::SoCRegisterFmt>,
+            ),
+            None, // LiteX simulator has no UART phy
+            dynamic_deferred_caller,
+        )
+    );
+    uart0.initialize(
+        dynamic_deferred_caller
+            .register(uart0)
+            .expect("dynamic deferred caller out of slots"),
+    );
+
+    PANIC_REFERENCES.uart = Some(uart0);
+
+    // Create a shared UART channel for the console and for kernel debug.
+    //
+    // The baudrate is ingnored, as no UART phy is present in the
+    // verilated simulation.
+    let uart_mux =
+        components::console::UartMuxComponent::new(uart0, 115200, dynamic_deferred_caller)
```

Comment:
Here the scenario is somewhat different. A few lines above I'm initializing the UART without PHY registers (passing a `None`). This means that all calls to `configure()` end up with an `ReturnCode::ENOSUPPORT` -- which makes sense, since this is a simulation after all.

This baud rate is therefore just a dummy value. We could replace it with `1` or some other value? I would've hoped the comment above explains this sufficiently. What do you think?

Reply:
Ah, I see. This is probably fine then.